src/drivers/safety_button/SafetyButton.cpp

```diff
@@ -95,8 +95,9 @@ SafetyButton::CheckButton()
 			_button_counter++;
 
 		} else if (_button_counter == CYCLE_COUNT) {
-			// change to disarmed state and notify
-			_safety_btn_off = false;
+			// do not switch safety off,
+			// as a button that is stuck in hardware
+			// could cause this
```

Comment:
Reset (_safety_btn_off = false) on disarm?

Reply:
That would need to depend on the circuit breaker. Wondering if it is worth the additional complexity. I'd prefer to have a simpler system == keep as is.

Reply:
> Reset (_safety_btn_off = false) on disarm?

Problem with this would be if you're using the safety switch functionality you'd suddenly need to walk up to the vehicle after every single flight while currently you can turn saftey off once and fly as much as you want.